esmvaltool/cmorizers/data/utilities.py

```diff
@@ -455,36 +467,33 @@ def fix_dim_coordnames(cube):
                 coord_type)
             continue
 
-        if coord_type == 'T':
-            coord.var_name = 'time'
-            coord.attributes = {}
-
-        if coord_type == 'X':
-            coord.var_name = 'lon'
-            coord.standard_name = 'longitude'
-            coord.long_name = 'longitude coordinate'
-            coord.units = Unit('degrees')
-            coord.attributes = {}
-
-        if coord_type == 'Y':
-            coord.var_name = 'lat'
-            coord.standard_name = 'latitude'
-            coord.long_name = 'latitude coordinate'
-            coord.units = Unit('degrees')
-            coord.attributes = {}
-
-        if coord_type == 'Z':
-            if coord.var_name == 'depth':
-                coord.standard_name = 'depth'
-                coord.long_name = \
-                    'ocean depth coordinate'
-                coord.var_name = 'lev'
-                coord.attributes['positive'] = 'down'
-            if coord.var_name == 'pressure':
-                coord.standard_name = 'air_pressure'
-                coord.long_name = 'pressure'
-                coord.var_name = 'air_pressure'
-                coord.attributes['positive'] = 'up'
+        coord.attributes = {}
+
+        # lookup table
+        lut = {
+            'T': 'time',
+            'X': 'longitude',
+            'Y': 'latitude',
+            'Z': {
+                'depth': ('depth_coord', 'down'),
+                # Any pressure coord would work.
```

Comment:
pXXX coordinates don't have a `positive` attribute because they are scalar coords set in a given level. The useful coordinates in here should be plevX ones. Depth and plev coord info tables have a a `stored_direction` value set to either increasing or decreasing that can be of use when setting this attribute.

Reply:
Updated to look for the first available plevX coord (as different X's are available in different projects) and to use the stored_direction value to set the positive attribute of the cube.
Let me know if that suits the requirements.

Reply:
One of the tests was failing, after investigation I noticed that to match the tests I have to assign the following value for `positive` attribute:
- 'down' if `stored_direction` == 'increasing'
- 'up' if `stored_direction` == 'decreasing'

Which sounds a bit counter-intuitive for me. Could you confirm me that this is the way to go, or the test is also wrong ?

I also noticed a `positive` value in the `_json_data` section of the CMOR_TABLE coord entries, but it is equal to 'down' for both depth and pressure coords.